Classes/Core/Html/HtmlParser.php

```diff
@@ -0,0 +1,328 @@
+<?php
+
+/*
+ * This file is part of the TYPO3 CMS project.
+ *
+ * It is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License, either version 2
+ * of the License, or any later version.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE.txt file that was distributed with this source code.
+ *
+ * The TYPO3 project - inspiring people to share!
+ */
+
+namespace TYPO3\CMS\FrontendEditing\Core\Html;
+
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+class HtmlParser extends \TYPO3\CMS\Core\Html\HtmlParser
```

Comment:
Add a comment with info about what's different in this class vs the parent. It looks like quite a massive rewrite, so if someone wants to look at it later it will be good to have a clear statement of intent.